tests/v2/validation/provisioning/rke2/encryption_key_rotation_test.go

```diff
@@ -0,0 +1,270 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	apiv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/secrets"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type RKE2EncryptionKeyRotationTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	config      *provisioning.Config
+	clusterName string
+	namespace   string
+}
+
+const scale = 10000
+
+var phases = []rkev1.RotateEncryptionKeysPhase{
+	rkev1.RotateEncryptionKeysPhasePrepare,
+	rkev1.RotateEncryptionKeysPhasePostPrepareRestart,
+	rkev1.RotateEncryptionKeysPhaseRotate,
+	rkev1.RotateEncryptionKeysPhasePostRotateRestart,
+	rkev1.RotateEncryptionKeysPhaseReencrypt,
+	rkev1.RotateEncryptionKeysPhasePostReencryptRestart,
+	rkev1.RotateEncryptionKeysPhaseDone,
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.config = new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, r.config)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clusterName = r.client.RancherConfig.ClusterName
+	r.namespace = r.client.RancherConfig.ClusterName
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TestEncryptionKeyRotationFreshCluster(provider Provider, kubeVersion string, nodesAndRoles []machinepools.NodeRoles, credential *cloudcredentials.CloudCredential) {
+	name := fmt.Sprintf("Provider_%s/Kubernetes_Version_%s/Nodes_%v", provider.Name, kubeVersion, nodesAndRoles)
+	r.Run(name, func() {
+		r.Run("initial", func() {
```

Comment:
Sorry if this is a stupid question, but what does this `initial` test label refer to? Are we planning on creating more test sub-suites alongside this `initial` one?

Reply:
I was originally going to separate the provisioning and the rotation, but scrapped that idea, so this "initial" test case can be unwrapped, good catch.